docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,795 @@
+ready messages, the ready protocol completes and the signing protocol starts.
+For liquidation recovery purposes, this ready protocol is the one that is
+adjusted. Instead of only announcing sender IDs, the three signers announce a
+ready message with three components:
+
+- The sender ID.
+- The BTC address to which the sender's share will be sent.
+- The max fee per vByte the sender is willing to pay.
+
+Once each client receives the ready messages from the two other nodes, it can
+immediately choose the correct fee per vByte from the provided values (see
+<<bitcoin-fee>>), construct the sighash (see <<sighash-construction>>) and
+proceed into the signing process to sign that sighash. Once a signature is
+computed, each client should construct the final transaction (see
+<<final-transaction-construction>>) and submit it (see
+<<transaction-submission>>).
+
+When receiving the ready message from other nodes, a node should validate that
+the announced BTC address can be parsed correctly by the tooling used to handle
+transaction and sighash construction. If the announced BTC address cannot be
+parsed, an error should be logged and the message should be treated as if it was
+not received.
+
+The retry mechanism for the signing process for liquidation recovery should be
+identical to regular signing, repeating every 2 minutes that a signature has
+not been completed successfully until the liquidation recovery timeout. Unlike
+standard signing, the liquidation recovery timeout should be very long to allow
+for node malfunctions to recover cleanly: this RFC recommends a default timeout
+of 7 days.
+
+[[transaction-submission]]
+==== Submitting the Signed Transaction
+
+Submitting a signed Bitcoin transaction requires a connection to a Bitcoin
+node, direct or indirect. To avoid _strictly requiring_ a connection to a
+Bitcoin node, this RFC proposes logging the signed transaction 5 times on the
+WARN level with a descriptive prefix. Current operator guidance suggests that
+more than 5 WARN level logs within 5 minutes should trigger a notification
+for closer monitoring by operators, ensuring these logs will be noticed. The
+descriptive prefix will further allow for targeted notifications by operators
+so that they can flag these logs for follow-up action.
+
+The suggested structure of the log message is:
+
+```
+2020-02-01T12:12:12.12-0800    WARN   keep-ecdsa   Please broadcast Bitcoin transaction 0010010101000828181124af234gbawed000
+```
+
+The first three components are the standard components emitted by the ECDSA
+client's logging library, while the message itself begins with `Please broadcast
+Bitcoin transaction`, followed by the hexadecimal string representing the signed
+transaction. The signed transaction can be broadcast via manual submission to a
+known node, via API access to a service provider like BlockCypher or
+Blockstream, or via a manual form submission on a page like
+https://live.blockcypher.com/btc/pushtx/ .
```

Comment:
Is there no way to broadcast this TX automatically even if `Extensions.TBTC.ElectrsURL` is set, correct?

Reply:
The tx can be broadcast automatically if there's an optional Bitcoin node connection, next paragraph describes it.